src/Http/WebUtilities/src/FileBufferingWriteStream.cs

```diff
@@ -142,20 +142,27 @@ public override void Write(byte[] buffer, int offset, int count)
         public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
             ThrowArgumentException(buffer, offset, count);
+            await WriteAsync(buffer.AsMemory(offset, count), cancellationToken);
+        }
+
+        /// <inheritdoc />
+        [SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads", Justification = "This is a method overload.")]
+        public override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken = default)
+        {
             ThrowIfDisposed();
 
-            if (_bufferLimit.HasValue && _bufferLimit - Length < count)
+            if (_bufferLimit.HasValue && _bufferLimit - Length < buffer.Length)
```

Comment:
Offset wasn't being used before when looking at these two checks. I'm going to assume that was a bug. @Tratcher

Reply:
Offset isn't important in this check, only the count/length. Offset was checked by ThrowArgumentException.